root/user/UserProfile.js

```diff
@@ -0,0 +1,537 @@
+          <UserProfileProperty name={l('Location:')}>
+            <DescriptiveLink entity={user.area} />
+          </UserProfileProperty>
+        ) : null}
+
+        <UserProfileProperty name={l('Member since:')}>
+          {memberSince}
+        </UserProfileProperty>
+
+        {$c.user && (viewingOwnProfile || $c.user.is_account_admin) ? (
+          <UserProfileProperty name={l('Last login:')}>
+            {user.last_login_date
+              ? formatUserDate($c.user, user.last_login_date)
+              : l("Hasn't logged in yet")}
+          </UserProfileProperty>
+        ) : null}
+
+        {user.website ? (
+          <UserProfileProperty name={l('Homepage:')}>
+            {showBioAndURL ? (
+              <a href={user.website} rel="nofollow">
+                {user.website}
+              </a>
+            ) : (
+              <div className="deleted">
+                {exp.l(
+                  `This content is hidden to prevent spam.
+                   To view it, please {url|log in}.`,
+                  {url: '/account/login'},
+                )}
+              </div>
+            )}
+          </UserProfileProperty>
+        ) : null}
+
+        <UserProfileProperty name={l('Subscribers:')}>
+          {subscriberCount ? (
+            exp.l(
+              '{count} ({url|view list})',
+              {count: subscriberCount, url: `/user/${encodedName}/subscribers`},
+            )
+          ) : (
+            l('0')
+          )}
+          {$c.user_exists && !viewingOwnProfile ? (
+            <>
+              {' '}
+              {bracketed(
+                subscribed ? (
+                  <a href={`/account/subscriptions/editor/remove?id=${encodedName}`}>
+                    {l('unsubscribe')}
+                  </a>
+                ) : (
+                  <a href={`account/subscriptions/editor/add?id=${encodedName}`}>
+                    {l('subscribe')}
+                  </a>
+                ),
+              )}
+            </>
+          ) : null}
+        </UserProfileProperty>
+
+        {user.biography ? (
+          <UserProfileProperty className="biography" name={l('Bio:')}>
+            {showBioAndURL ? (
+              expand2react(user.biography)
+            ) : (
+              <div className="deleted">
+                {exp.l(
+                  `This content is hidden to prevent spam.
+                   To view it, please {url|log in}.`,
+                  {url: '/account/login'},
+                )}
+              </div>
+            )}
+          </UserProfileProperty>
+        ) : null}
+
+        {user.languages && user.languages.length > 0 ? (
+          <UserProfileProperty name={l('Languages:')}>
+            <ul className="inline">
+              {user.languages.map(language => (
+                <li key={language.language.id}>
+                  {l_languages(language.language.name)}
+                  {' '}
+                  {bracketedText(l(capitalize(language.fluency)))}
```

Comment:
This is a faithful rewrite of the original Perl subroutine `l_fluency` that was calling `ucfirst` but it was/is still hackish. It only works because fluency’s labels can be programmatically deduced from associated text values. Maybe that trick should be documented in https://github.com/metabrainz/musicbrainz-server/blob/v-2019-08-22/lib/MusicBrainz/Server/Form/User/EditProfile.pm#L70-L77 so as to avoid breaking user profile when renaming one of these options.

Reply:
It duplicates [`options_languages_fluency`](https://github.com/metabrainz/musicbrainz-server/blob/62c2a79bc1db08fe0a40240e75d84c42ddc94bac/lib/MusicBrainz/Server/Form/User/EditProfile.pm#L70-L77) now.

Reply:
Hmm. Can we reuse fluencyOptions from EditProfileForm somehow for this instead? (Guess that'd be easier than passing it from an unrelated Perl file, anyway).

Reply:
It is potentially used by `FormHandler`, otherwise probably yes.